acme/acme/messages.py

```diff
@@ -134,12 +136,43 @@ def code(self) -> Optional[str]:
             return code
         return None
 
+    @property
+    def codes(self) -> Optional[List[str]]:
+        """List of ACME error codes for error and all subproblems.
+
+        :returns: list of error codes if standard ACME codes or ``None``.
+        :rtype: List[str]
+        """
+        codes = []
+        subproblems = []
+        if self.subproblems is not None:
+            subproblems = self.subproblems
+        if self.code is not None:
+            codes.append(self.code)
+
+        for error in subproblems:
+            if error.code:
+                codes.append(error.code)
+        if len(codes) > 0:
+            return codes
+        return None
+
     def __str__(self) -> str:
-        return b' :: '.join(
+        subproblems = []
+        if self.subproblems is not None:
+            subproblems = self.subproblems
```

Comment:
This was originally added to silence an error in `tox -e lint` when iterating over `self.subproblems`, as it could previously be `None` which is not iterable.

I tried removing this in the latest version, and it seems to work fine now though...